presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainOrcPredicate.java

```diff
@@ -185,8 +187,14 @@ public boolean matches(long numberOfRows, Map<Integer, ColumnStatistics> statist
             }
         }
 
+        // did we read any bloom filter?
+        if (numberOfBloomFiltersChecked < 1) {
```

Comment:
It doesn't, the edge case it covers basically is filtering on partitioned columns (the unit test that failed in my initial PR). In addition it has the benefit of being able to debug log how many we had to check before being sure, can help to optimize on which columns you enable bloomfilters and even how to write your queries.